src/lib/network_pool/indexed_pool.ml

```diff
@@ -1552,7 +1557,16 @@ let%test_module _ =
                       !"Expired user command. Current global slot is \
                         %{sexp:Mina_numbers.Global_slot.t} but user command is \
                         only valid until %{sexp:Mina_numbers.Global_slot.t}"
-                      global_slot_since_genesis valid_until () )
+                      global_slot_since_genesis valid_until ()
+                | Error
+                    (Expired
+                      ( `Timestamp_predicate
+                      , `Global_slot_since_genesis global_slot_since_genesis ))
+                  ->
+                    failwithf
+                      !"Expired zkapp. Current global slot is \
```

Comment:
could you include `expiry_ns` value to this message?  Maybe something like `Transaction expired or will expire in the pool based on the current expiry duration of %d`